components/sup/src/event.rs

```diff
@@ -233,14 +238,19 @@ pub fn health_check(metadata: ServiceMetadata,
         let StandardStreams { stdout, stderr } =
             maybe_process_output.map(ProcessOutput::standard_streams)
                                 .unwrap_or_default();
+
+        let time_interval: Option<Duration> = health_check_interval.into();
+        let prost_interval = prost_duration::from(time_interval.unwrap());
```

Comment:
Using the [`From` trait here](https://github.com/habitat-sh/habitat/blob/c1e104258c73ddbace9c567deda347f46583dfac/components/core/src/service.rs#L405), this could be better expressed as:

```rust
let prost_interval = prost_duration::from(Duration::from(health_check_interval));
```